content/src/components/editor/rx/editor.ts

```diff
@@ -60,43 +65,76 @@ export const editorRx = Rx.family((handle: RxWorkspaceHandle) => {
           const path = workspace.fullPathTo(file)
           return Option.match(path, {
             onNone: () => Effect.void,
-            onSome: (path) => container.writeFile(path, editor.editor.getValue(), "typescript")
+            onSome: (path) =>
+              container.writeFile(
+                path,
+                editor.editor.getValue(),
+                "typescript"
+              )
           })
         })
       )
 
+      /**
+       * Saves the content of the editor's current model to the local storage.
+       */
+      const startAutoSave = editor.content.pipe(
+        Stream.debounce("2 seconds"),
+        Stream.runForEach((_) =>
+          Effect.gen(function* () {
+            const workspace = yield* SubscriptionRef.get(handle.workspace)
+            return yield* autosave.saveToLocal(workspace, "save")
+          })
+        ),
+        Effect.forkScoped
+      )
+
       /**
        * Syncs the content of the editor with the underlying WebContainer file
        * system.
        */
       function sync(fullPath: FullPath, file: File) {
         return container.readFile(fullPath).pipe(
           Stream.tap((model) => editor.loadModel(model)),
-          Stream.flatMap(() => editor.content.pipe(Stream.drop(1)), { switch: true }),
+          Stream.flatMap(() => editor.content.pipe(Stream.drop(1)), {
+            switch: true
+          }),
           Stream.debounce("2 seconds"),
-          Stream.tap((content) => container.writeFile(fullPath, content, file.language)),
-          Stream.ensuring(Effect.suspend(() => {
-            const content = editor.editor.getValue()
-            if (content.trim().length === 0) {
-              return Effect.void
-            }
-            return container.writeFile(fullPath, content, file.language)
-          }))
+          Stream.tap((content) =>
+            container.writeFile(fullPath, content, file.language)
+          ),
+          Stream.ensuring(
+            Effect.suspend(() => {
+              const content = editor.editor.getValue()
+              if (content.trim().length === 0) {
+                return Effect.void
+              }
+              return container.writeFile(fullPath, content, file.language)
+            })
+          )
```

Comment:
@titouancreach - there are a lot of these unnecessary formatting changes in this PR, which makes it difficult to parse what the actual changes to the codebase are.

Could you please remove an ancillary formatting changes?

Reply:
Yes sure ! I used format and save with prettier since I have seen a prettierrc file !